tests/CMakeLists.txt

```diff
@@ -76,3 +82,20 @@ endif()
 
 include_directories(${CMAKE_SOURCE_DIR}/src ${PROJECT_BINARY_DIR})
 configure_file("${PROJECT_SOURCE_DIR}/tests/config.h.in" "${PROJECT_BINARY_DIR}/tests/config.h" ESCAPE_QUOTES @ONLY)
+
+if(LIBPAM_HAVE_CONFDIR)
+
+    #compile PAM test module
+    set(CMAKE_SHARED_LIBRARY_PREFIX  )
```

Comment:
This removes the library prefix globally for the whole CMake list and any nested ones. While being fine in this case, it could cause problems later and generally it is best to set similar options per-target. Use `SET_TARGET_PROPERTIES(name PROPERTIES PREFIX "")` with `name` being the target.